src/clj/nnts2/middleware.clj

```diff
@@ -63,18 +64,19 @@
     ([request respond raise]
      (respond (handler request)))))
 
-(defn debug [x] (prn x) x)
-
 (defn wrap-validate-access-token [handler]
   (fn [request]
-    (let [access-token (get-in request [:session :ring.middleware.oauth2/access-tokens :google :token])]
-      (try (let [response (clj-http/get "https://www.googleapis.com/oauth2/v3/userinfo"
-                                        {:headers {"Authorization" (str "Bearer " access-token)}} {:as :json :throw-exceptions false})
-                 user-info (-> response
-                               (:body)
-                               (json/parse-string true)
-                               (hyphenize-collection))]
-             (handler (assoc-in request [:session :user-info] user-info)))
-           (catch Exception e
-             (-> (res/response "Not authorized")
-                 (res/status 401)))))))
+    (let [access-token (or (get-in request [:session :ring.middleware.oauth2/access-tokens :google :token])
+                           (get-in request [:headers "authorization"]))]
```

Comment:
When are we picking google access-token from headers?

Reply:
When running tests on Postman.